centreon-certified/omi/omi_events-apiv2.lua

```diff
@@ -0,0 +1,363 @@
+--
+-- Copyright 2022 Centreon
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+-- For more information : contact@centreon.com
+--
+-- To work you need to provide to this script a Broker stream connector output configuration
+-- with the following informations:
+--
+-- source_ci (string): Name of the transmiter, usually Centreon server name
+-- ipaddr (string): the ip address of the operation connector server
+-- url (string): url of the operation connector endpoint
+-- logfile (string): the log file to use
+-- loglevel (number): th log level (0, 1, 2, 3) where 3 is the maximum level
+-- port (number): the operation connector server port
+-- max_size (number): how many events to store before sending them to the server.
+-- max_age (number): flush the events when the specified time (in second) is reach (even if max_size is not reach).
+
+-- Libraries
+local curl = require("cURL")
+local http = require("socket.http")
+local ltn12 = require("ltn12")
+
+-- Centreon lua core libraries
+local sc_common = require("centreon-stream-connectors-lib.sc_common")
+local sc_logger = require("centreon-stream-connectors-lib.sc_logger")
+local sc_broker = require("centreon-stream-connectors-lib.sc_broker")
+local sc_event = require("centreon-stream-connectors-lib.sc_event")
+local sc_params = require("centreon-stream-connectors-lib.sc_params")
+local sc_macros = require("centreon-stream-connectors-lib.sc_macros")
+local sc_flush = require("centreon-stream-connectors-lib.sc_flush")
+
+-- workaround https://github.com/centreon/centreon-broker/issues/201
+local previous_event = ""
+
+--------------------------------------------------------------------------------
+-- EventQueue class
+--------------------------------------------------------------------------------
+
+local EventQueue = {}
+EventQueue.__index = EventQueue
+
+--------------------------------------------------------------------------------
+-- Constructor
+-- @param conf The table given by the init() function and returned from the GUI
+-- @return the new EventQueue
+--------------------------------------------------------------------------------
+function EventQueue.new(params)
+  local self = {}
+  self.fail = false
+
+  local mandatory_parameters = {
+      "ipaddr",
+      "url",
+      "port"
+  }
+
+  -- set up log configuration
+  local logfile = params.logfile or "/var/log/centreon-broker/omi_event.log"
+  local log_level = params.log_level or 2
```

Comment:
```suggestion
  local log_level = params.log_level or 1
```

never put 2 as a default value. It talks too much for nothing unless you want to debug your stream connector